src/relentless/model/potential/bond.py

```diff
@@ -0,0 +1,425 @@
+import numpy
+
+from relentless import math
+
+from . import potential
+
+
+class BondParameters(potential.Parameters):
+    """Parameters for a bond potential."""
+
+    pass
+
+
+class BondPotential(potential.Potential):
+    r"""Abstract base class for a bond potential."""
+
+    pass
+
+
+class Harmonic(BondPotential):
+    r"""Harmonic bond potential.
+
+    .. math::
+
+        u(r) = \frac{k}{2} (r - r_0)^2
+
+    where :math:`r` is the distance between two bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+-----------+
+    | Parameter   | Description                                      | Initial   |
+    +=============+==================================================+===========+
+    | ``k``       | potential constant                               |           |
+    +-------------+--------------------------------------------------+-----------+
+    | ``r0``      | rest length.                                     |           |
+    +-------------+--------------------------------------------------+-----------+
+
+    Parameters
+    ----------
+    type : str
+        Type.
```

Comment:
Check how this is documented in pair, but this potential should support multiple bond types.
```suggestion
    types: tuple[str]
        Types.
```